python/cudf/cudf/utils/utils.py

```diff
@@ -76,22 +76,35 @@ def require_writeable_array(arr):
     return np.require(arr, requirements="W")
 
 
-def scalar_broadcast_to(scalar, shape, dtype):
+def scalar_broadcast_to(scalar, shape, dtype=None):
     from cudf.utils.cudautils import fill_value
-    from cudf.utils.dtypes import to_cudf_compatible_scalar
+    from cudf.utils.dtypes import to_cudf_compatible_scalar, is_string_dtype
+    from cudf.core.column import column_empty
 
-    scalar = to_cudf_compatible_scalar(scalar, dtype=dtype)
+    if scalar is None:
+        if dtype is None:
+            dtype = "object"
+        return column_empty(np.prod(shape), dtype=dtype, masked=True)
```

Comment:
don't use `np.prod` it's slow as you've solved for Numba before 😉

Reply:
Fixed

Reply:
Actually, not fixed. I decided to go with `prod` as a matter of convenience. If `shape` is an integer-like, we'd have to explicitly convert it to a `tuple` before calling `reduce`. Relative to the time for doing a scalar broadcast, I'm hoping that `np.prod` will be negligible.

Let me know if you still prefer `reduce`

Reply:
Did a quick test:
```
In [40]: def test_reduce(a):
    ...:     if isinstance(a, (int, np.integer)):
    ...:         return a
    ...:     elif isinstance(a, collections.Iterable):
    ...:         return functools.reduce(lambda x, y: x*y, a)
    ...:     else:
    ...:         raise ValueError
    ...:

In [41]: %timeit test_reduce(10)
209 ns ± 0.677 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [42]: %timeit test_reduce(10)
210 ns ± 1.15 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [43]: %timeit test_reduce((10,))
1.11 µs ± 1.36 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [44]: %timeit np.prod(10)
4.17 µs ± 40.8 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [45]: %timeit np.prod((10,))
4.69 µs ± 52.3 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

For scalars it's more than an order of magnitude faster, and for smaller iterables it's ~4x faster. I think it's worth it, especially for the scalar case.